### PR TITLE
Mock data added on test_fermi_gbm

### LINKED ISSUE #2874: Work towards mocking all the online tests in SunPy

### PR DESCRIPTION
**Pull request by darthgera123**
### Description

Mocking fermi testing. Currently a WIP.

Partially Fixes #2874

### CODE REVIEW

**Comment by nabobalis** [comment]
You will need to update `astropy_helpers`, it seems you had the old version and added it by mistake. 
Hopefully doing:
```
 git submodule update --init
```
should revert those changes (assuming you have an up to date remotes)

**Comment by darthgera123** [comment]
please review @nabobalis

File: sunpy/net/dataretriever/tests/test_fermi_gbm.py [hunk 1]
```diff
@@ -46,24 +97,30 @@ def test_can_handle_query(time):
...
-    (a.Time('2012/11/27', '2012/11/27'), a.Instrument('gbm')),
-])
-def test_get(time, instrument):
-    qr1 = LCClient.search(time, instrument)
+# @pytest.mark.remote_data
+# @pytest.mark.parametrize("time,instrument", [
+#     (a.Time('2012/11/27', '2012/11/27'), a.Instrument('gbm')),
+# ])
+# def test_get(time, instrument):
+#     qr1 = LCClient.search(time, instrument)
```
Comment (nabobalis) [inline sunpy/net/dataretriever/tests/test_fermi_gbm.py]: Could you remove the commented code?

File: sunpy/net/dataretriever/tests/test_fermi_gbm.py [hunk 2]
```diff
@@ -46,24 +97,30 @@ def test_can_handle_query(time):
...
+#     download_list = LCClient.fetch(qr1)
+#     assert len(download_list) == len(qr1)
+
+@mock.patch('sunpy.net.dataretriever.sources.fermi_gbm.GBMClient.search',
+            return_value=mock_querry_object('2012/11/27', '2012/11/27'))
+def test_get(time):
+    qr1 = LCClient.search(time, a.Instrument('gbm'))
```
Comment (nabobalis) [inline sunpy/net/dataretriever/tests/test_fermi_gbm.py]: I'm not sure this test makes any sense as a mock test. But I don't have a better idea.

File: sunpy/net/dataretriever/tests/test_fermi_gbm.py (hunk 1 above)
Comment (darthgera123) [reply sunpy/net/dataretriever/tests/test_fermi_gbm.py]: yeah, sure

File: sunpy/net/dataretriever/tests/test_fermi_gbm.py (hunk 2 above)
Comment (darthgera123) [reply sunpy/net/dataretriever/tests/test_fermi_gbm.py]: what do you mean? Like theres no point in mocking this test?

Comment (nabobalis) [reply sunpy/net/dataretriever/tests/test_fermi_gbm.py]: Oh I got the wrong test, when I was looking at the diff.

**Pull request closed without merging**